source/rafcon/gui/start.py

```diff
@@ -199,14 +209,46 @@ def setup_argument_parser():
     parser.add_argument('-s', '--start_state_path', metavar='path', dest='start_state_path', default=None, nargs='?',
                         help=_("path within a state machine to the state that should be launched which consists of "
                                "state ids e.g. QPOXGD/YVWJKZ where QPOXGD is the root state and YVWJKZ its child states"
-                               " to start from."))
+                               " to start from"))
     parser.add_argument('-q', '--quit', dest='quit_flag', action='store_true',
                         help=_("a flag to specify if the gui should quit after launching a state machine"))
+    parser.add_argument('-mp', '--memory-profiling', dest='memory_profiling', action='store_true',
+                        help=_("a flag to specify if the gui should enable memory profiling"))
+    parser.add_argument('-mpp', '--memory-profiling-path', action='store', type=config_path, metavar='path', dest='memory_profiling_path',
+                        default=default_log_path, nargs='?', const=default_log_path,
+                        help=_("path to the memory profiling log memoy_profiling.log").format(default_log_path))
+    parser.add_argument('-mpi', '--memory-profiling-interval', dest='memory_profiling_interval', action='store', default=10,
+                        help=_("The interval between snapshots creaton for memory profiling in seconds"))
+    parser.add_argument('-mppr', '--memory-profiling-print', dest='memory_profiling_print', action='store_true',
+                        help=_("a flag to specify if the memory profiling results should be printed"))
+
     return parser
 
 
-def setup_mvc_configuration(core_config_path, gui_config_path, runtime_config_path):
+def setup_mvc_configuration(core_config_path, gui_config_path, runtime_config_path, design_config):
+    """ Loads all configurations from disk
+
+    :param core_config_path: the path to the core configuration file
+    :param gui_config_path:  the path to the gui configuration file
+    :param runtime_config_path:  the path to the runtime configuration file
+    :param design_config:  the path to the design configuration file
+    :return:
+    """
     setup_configuration(core_config_path)
+    # the design config has to be loaded before loading the gui config as it is used by the gui config
+    if design_config:
+        design_config_path, design_config_file = filesystem.separate_folder_path_and_file_name(design_config)
+        global_design_config.load(design_config_file, design_config_path)
+        # the constant has to be overwritten here,
+        # as the singleton was already loaded that hus the wrong INTERFACE_FONT was chosen
+        from rafcon.gui.utils import constants
+        constants.INTERFACE_FONT = global_design_config.get_config_value("PRIMARY_FONT")
```

Comment:
Why is this done only for design passed via the command line and not those set through the env var? I think th if and else branch can be merged.